### PR TITLE
FIT-412 Build Deploy Failure Slack Notification

### PR DESCRIPTION
**Pull request by osi-richbach**

### CODE REVIEW

**Comment by osi-richbach** [comment]
retest this please

File: src/gov/ca/cwds/jenkins/SlackNotifier.groovy [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  def script
+  def webhookUrl
+
+  SlackNotifier(script, webhookUrl) {
+    this.script = script
+    this.webhookUrl = webhookUrl
+  }
+
+  def notifySlack(projectName, error) {
+    def payload = """{
```
Comment (edgibbs) [inline src/gov/ca/cwds/jenkins/SlackNotifier.groovy]: Inside a triple quoted string you should be able to use single and double quotes without escaping, I pulled them and all lint/tests passed.

File: test/gov/ca/cwds/jenkins/SlackNotifierSpecification.groovy [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    def curlCommand = "curl -X POST -H 'Content-type: application/json' --data '${payload}' http://exampleslack.url"
+    pipeline.sh( [script: curlCommand]) >> 'ok'
+
+    def slackNotifier = new SlackNotifier(pipeline, 'http://exampleslack.url')
+
+    when:
+    def returnValue = slackNotifier.notifySlack('myproject', error)
+
+    then:
+    returnValue == 'ok'
```
Comment (edgibbs) [inline test/gov/ca/cwds/jenkins/SlackNotifierSpecification.groovy]: OK, this was a rabbit hole, but I believe just invoking script.sh will return null/void.  Therefore this test is testing something that can't happen in the real production code.  Probably better to just assert that pipeline.sh was called with the correct argument.

Comment (osi-richbach) [reply test/gov/ca/cwds/jenkins/SlackNotifierSpecification.groovy]: good call....will take care of this

File: src/gov/ca/cwds/jenkins/SlackNotifier.groovy (hunk 1 above)
Comment (osi-richbach) [reply src/gov/ca/cwds/jenkins/SlackNotifier.groovy]: I must be doing something wrong....string interpolation doesn't seem to work if I switch.

Comment (osi-richbach) [reply src/gov/ca/cwds/jenkins/SlackNotifier.groovy]: forget it....I'm slow to the party but got it now.

**Pull request merged**